composer/models/huggingface.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from torch import nn
+from torchmetrics import Metric
+from torchmetrics.collections import MetricCollection
+
+from composer.models.base import ComposerModel
+
+if TYPE_CHECKING:
+    import transformers
+
+__all__ = ["HuggingFaceModel"]
+
+
+class HuggingFaceModel(ComposerModel):
+    """
+    Args:
+        model ([`PreTrainedModel`] or `torch.nn.Module`):  You can  use your own models defined as
+        `torch.nn.Module` as long as they work the same way as the 🤗 Transformers models.
```

Comment:
```suggestion
        :class:`torch.nn.Module` as long as they work the same way as the 🤗 Transformers models.
```

Reply:
Can we specify more about "as long as they work the same way"? What minimal set of methods do we require to be replicated?

Reply:
same as i said to davis: how would you feel about limiting it to just huggingface pretrained models. I lifted this from the huggingface trainer docs, but looking at it now i feel it makes less sense for us. non HF models probably shouldn't use the HF integration class.